src/data_write.py

```diff
@@ -540,56 +539,69 @@ def __init__(self, data_write_options):
         self.first_time = True
 
     @staticmethod
-    def write_json_file(filename, data_dict):
+    def write_json_file(filename, slice_data, file_type):
         """
         Write out data to a json file. If the file already exists it will be overwritten.
 
         :param  filename:   The path to the file to write out
         :type   filename:   str
-        :param  data_dict:  Python dictionary to write out to the JSON file.
-        :type   data_dict:  dict
+        :param  slice_data: Data to write out to the JSON file.
+        :type   slice_data: SliceData
+        :param  file_type:  The type of file to write.
+        :type   file_type:  str
         """
+        data_dict = {}
+        for relevant_field in SliceData.type_fields(file_type):
+            data_dict[relevant_field] = getattr(slice_data, relevant_field)
 
         with open(filename, 'w+') as f:
             f.write(json.dumps(data_dict))
 
     @staticmethod
-    def write_hdf5_file(filename, data_dict, dt_str):
+    def write_hdf5_file(filename, slice_data, dt_str, file_type):
         """
         Write out data to an HDF5 file. If the file already exists it will be overwritten.
 
         :param  filename:   The path to the file to write out
         :type   filename:   str
-        :param  data_dict:  Python dictionary to write out to the HDF5 file.
-        :type   data_dict:  dict
+        :param  slice_data: Data to write out to the HDF5 file.
+        :type   slice_data: SliceData
         :param  dt_str:     A datetime timestamp of the first transmission time in the record
         :type   dt_str:     str
+        :param  file_type:  Type of file to write.
+        :type   file_type:  str
         """
 
-        def convert_to_numpy(dict_of_lists):
-            """
-            Converts lists stored in dict into numpy array. Recursive.
-
-            :param  dict_of_lists: Dictionary with lists to convert to numpy arrays.
-            :type   dict_of_lists: dict
-            """
-            for k, v in dict_of_lists.items():
-                if isinstance(v, dict):
-                    convert_to_numpy(v)
-                elif isinstance(v, list):
-                    dict_of_lists[k] = np.array(v)
-                else:
-                    continue
-
-        convert_to_numpy(data_dict)
-
-        time_stamped_dd = {dt_str: data_dict}
-
-        # Ignoring warning that arises from using integers as the keys of the data dictionary.
-        warnings.simplefilter('ignore', tables.NaturalNameWarning)
-
         try:
```

Comment:
From testing this on LAB we saw that something is not quite right in this following block when integrating with pyDARNio. Can we revery 'data_write.py' and push the rest of this PR through, then fix this up when pyDARNio is patched?